### ISSUE TITLE
Improvement for k8s.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/

### ISSUE THREAD

**Issue by dwgillies-bluescape** [kubernetes/website#35389]
The HPA walkthrough suggests I learn about metrics server, in case I am using minikube.  No.  That is unhelpful.  minikube has its own internal metrics server.  I tried installing 2 outside metrics servers before I gave up and realized that doesn't work.  You should not send the learning in the right direction in the introduction to HPA.  Please fix the language on this page as follows.  I enclose the old line, and a new replacement line : 

- To learn how to deploy the Metrics Server, see the [metrics-server documentation](https://github.com/kubernetes-sigs/metrics-server#deployment).

- To learn how to deploy the Metrics Server, see the [metrics-server documentation](https://github.com/kubernetes-sigs/metrics-server#deployment).  If you are using minikube, learn how to [enable the metrics server in minikube](https://kubernetes.io/docs/tutorials/hello-minikube/#enable-addons).  Also, see https://github.com/kubernetes/minikube/issues/13620 to get cpu metrics to work with minikube.

omg minikube is a POS isn't it?

===========

Also, when I ran the HPA walkthrough, just because you list the steps, doesn't mean that the metrics-server is up and producing statistics that HPA can read.  For example, once I enabled metrics-server, my HPA was still not working - the declaration had to be bounced before HPA would actually gather metrics from my new metrics server.  You should tell th learner to describe the metrics server after it's installed to make sure its producing metrics that hpa can read.  If kubectl get hpa -n apache-php prints out something like this, it means that the metrics server is not producing valid results :

NAME         REFERENCE               TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
php-apache   Deployment/php-apache   <unknown>/50%   1         10        0          2s

You should have them do an API call to metrics server to verify that the server is working properly : 

`kubectl get --raw "/apis/metrics.k8s.io/v1beta1/namespaces/default/pods" | jq`

**Comment by shannonxtreme** [comment]
I think point 2 is good, have a verify checkpoint before proceeding

For the minikube-specific instructions, I'd like to see us not cater to a specific Kubernetes implementation by name. So instead of linking to minikube specific docs, something like

"Some Kubernetes implementations, like minikube, have a built-in metrics-server. If manual metrics-server installation fails, refer to your Kubernetes product's documentation to check and enable the built-in metrics-server"

I'd definitely avoid referring to specific minikube troubleshooting in our docs. That's out of scope for this doc.

**Comment by dwgillies-bluescape** [comment]
I did get everything working and autoscaling working with minikube, but it was a huge hassle :
```
minikube start --extra-config=kubelet.housekeeping-interval=10s
minikube addons enable metrics-server
```
i can see now why people recommend kind or some other kubernetes emulators.

**Comment by divya-mohan0209** [comment]
I agree with what @shannonxtreme is saying wrt scope of the documentation.

**Comment by divya-mohan0209** [comment]
@dwgillies-bluescape Would you be able to retitle to provide the context of what exactly needs to be improved for contributors and also add a short description incorporating Shannon's suggestion above? We can then open it for contributions. Thank you!
